benchmarks/src/main/scala/benchmark/MergeDataLoad.scala

```diff
@@ -0,0 +1,197 @@
+  }
+
+  /**
+   * Creates the target table and all source table configuration used in merge test cases.
+   */
+  def runInternal(): Unit = {
+    val dbName = conf.dbName
+    val dbLocation = conf.dbLocation(dbName, suffix = benchmarkId.replace("-", "_"))
+    val dbCatalog = "spark_catalog"
+
+    require(conf.scaleInGB > 0)
+    require(Seq(1, 3000).contains(conf.scaleInGB), "")
+
+    log(s"====== Creating database =======")
+    runQuery(s"DROP DATABASE IF EXISTS ${dbName} CASCADE", s"drop-database")
+    runQuery(s"CREATE DATABASE IF NOT EXISTS ${dbName}", s"create-database")
+
+    log(s"====== Creating target table =======")
+    loadTargetTable()
+    log(s"====== Creating source tables =======")
+    MergeTestCases.testCases.map(_.sourceTable).distinct.foreach(loadSourceTable)
+    log(s"====== Created all tables in database ${dbName} at '${dbLocation}' =======")
+
+    runQuery(s"USE $dbCatalog.$dbName;")
+    runQuery("SHOW TABLES", printRows = true)
+  }
+
+  /**
+   * Creates the target Delta table and performs sanity checks.
+   */
+  protected def loadTargetTable(): Unit = {
+    val dbLocation = conf.dbLocation(conf.dbName, suffix = benchmarkId.replace("-", "_"))
+    val location = s"${dbLocation}/${conf.tableName}/"
+    val sourceFormat = "parquet"
+
+    runQuery(s"DROP TABLE IF EXISTS $targetTableFullName", s"drop-table-$targetTableFullName")
+
+    runQuery(
+      s"""CREATE TABLE $targetTableFullName
+                 USING DELTA
+                 LOCATION '$location'
+                 SELECT * FROM `${sourceFormat}`.`$dataSourceLocation`
+              """, s"create-table-$targetTableFullName", ignoreError = true)
+
+    val sourceRowCount =
+      spark.sql(s"SELECT * FROM `${sourceFormat}`.`$dataSourceLocation`").count()
+    val targetRowCount = spark.table(targetTableFullName).count()
+    val targetFileCount =
+      spark.table(targetTableFullName).select(countDistinct("_metadata.file_path"))
+    log(s"Target file count: $targetFileCount")
+    log(s"Target row count: $targetRowCount")
+
+    assert(targetRowCount == sourceRowCount,
+      s"Row count mismatch: source table = $sourceRowCount, " +
+      s"target $targetTableFullName = $targetRowCount")
+  }
+
+  /**
+   * Creates a source table for the given source table configuration by sampling the target table.
+   */
+  protected def loadSourceTable(sourceTable: MergeSourceTable): Unit = {
+    val fullTableName = s"`${conf.dbName}`.`${sourceTable.name}`"
+    val dbLocation = conf.dbLocation(conf.dbName, suffix = benchmarkId.replace("-", "_"))
+
+    runQuery(s"DROP TABLE IF EXISTS $fullTableName", s"drop-table-${sourceTable.name}")
+
+    val deltaDf = spark.read.format("delta")
+      .load(s"${dbLocation}/${conf.tableName}/")
+      // Sample files by hashing their paths.
+      .filter(hash(col("_metadata.file_path")) * 0.5 / Integer.MAX_VALUE + 0.5 < sourceTable.fileMatchedFraction)
```

Comment:
Why 0.5? Add a comment why this computation works.

Reply:
I've reworked the way I'm sampling files to make it less obscure and rely on spark's sample method instead,